scripts/apps/master-desk/MasterDesk.tsx

```diff
@@ -116,6 +116,7 @@ export class MasterDesk extends React.Component<{}, IState> {
                 filters[key] = null;
             }
         }
+        this.setState({filters: {}});
```

Comment:
If you are setting filters to an empty object, you shouldn't need the loop above?

Reply:
Since there's a reference of the Filter panel component `state.filter` in the parent component, I created a shallow copy of parent components and then changed the object keys' value. Which eventually updated the Filter panel component state.filter. So, using the single parent component function inside one child, I was able to clear the state of another. But yes, that was the bad approach, so I've changed it now.